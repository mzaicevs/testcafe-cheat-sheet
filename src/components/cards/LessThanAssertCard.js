import React from "react";
import { assert_lessThan } from "../../snippets";
import CodeCard from "../../base_components/CodeCard";

const LessThanAssertCard = () => (
  <CodeCard
    cardTitle={assert_lessThan.title}
    fullDocLink={assert_lessThan.fullDocLink}
    codeSnippet={{
      code: assert_lessThan.code,
      language: assert_lessThan.language,
      options: assert_lessThan.options,
      example: assert_lessThan.example
    }}
    description={assert_lessThan.description}
  />
);

export default LessThanAssertCard;
