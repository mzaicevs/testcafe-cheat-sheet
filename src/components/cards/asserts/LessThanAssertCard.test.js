import React from "react";
import { shallow } from "enzyme";
import LessThanAssertCard from "./LessThanAssertCard";

jest.mock("react-syntax-highlighter/dist/esm/styles/hljs", () => () => {});

describe("LessThanAssertCard", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<LessThanAssertCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
