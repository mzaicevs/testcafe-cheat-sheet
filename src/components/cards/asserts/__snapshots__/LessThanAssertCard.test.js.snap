// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LessThanAssertCard matches snapshot 1`] = `
<CodeCard
  cardTitle="Less Than - Assert"
  codeSnippet={
    Object {
      "code": Array [
        "t.expect( actual ).lt( expected, message, options );",
      ],
      "example": "import { Selector } from 'testcafe';

  fixture \`My fixture\`;
  
  test('My test', async t => {
      await t
        .expect(Selector('#element').offsetHeight).lt(25)
        .expect(2).lt(5, '2 is less than 5');
  });",
      "language": "javascript",
      "options": "{ 
    timeout: Number,
      allowUnawaitedPromise: Boolean 
  }",
    }
  }
  description="Asserts that actual is less than expected."
  fullDocLink="https://devexpress.github.io/testcafe/documentation/test-api/assertions/assertion-api.html#less-than"
/>
`;
